docs/luigi-core-api.md

```diff
@@ -0,0 +1,192 @@
+# Luigi Core API
+
+## Luigi.config()
+
+<!-- Generated by documentation.js. Update this documentation by updating the source code. -->
+
+### Configuration
+
+#### setConfig
+
+Sets the configuration for Luigi initially. Can also be called at a later point in time again to update the configuration.
+
+##### Parameters
+
+-   `configInput` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)?** the Luigi Core configuration object
+
+##### Examples
+
+```javascript
+Luigi.config().setConfig({
+  navigation: {
+    nodes: () => [
+      {
+        pathSegment: 'home',
+        label: 'Home',
+        children: [
+          {
+            pathSegment: 'hello',
+            label: 'Hello Luigi!',
+            viewUrl: '/assets/basicexternal.html'
+          }
+        ]
+      }
+    ]
+  },
+  routing: {
+    useHashRouting: true
+  }
+})
+```
+
+#### getConfigValue
+
+Gets value of the given property on Luigi config object. Target can be a value or a synchronous function.
+
+##### Parameters
+
+-   `property` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?** the object traversal path
+
+##### Examples
+
+```javascript
+Luigi.config().getConfigValue('auth.use')
+Luigi.config().getConfigValue('settings.sideNavFooterText')
+```
+
+#### getConfigBooleanValue
+
+Gets boolean value of the given property on Luigi config object.
+Function return true if the property value is equal true or 'true'. Otherwise the function returns false.
+
+##### Parameters
+
+-   `property` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?** the object traversal path
+
+##### Examples
+
+```javascript
+Luigi.config().getConfigBooleanValue('settings.hideNavigation')
+```
+
+#### getConfigValueAsync
+
+Gets value of the given property on the Luigi config object.
+If the value is a Function it is called (with the given parameters) and the result of that call is the value.
+If the value is not a Promise it is wrapped to a Promise so that the returned value is definitely a Promise.
+
+##### Parameters
+
+-   `property` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?** the object traversal path
+-   `parameters` **mixed?** optional parameters that are used if the target is a function
```

Comment:
Please remove the question mark at the end (by removing the square brackets around the parameter name in the source file).